### PR TITLE
added flux and scan methods; some refactoring

### PR DESCRIPTION
**Pull request by Megaprog**

### CODE REVIEW

File: src/main/java/com/appmetr/cql/CqlDao.java [hunk 1]
```diff
@@ -439,45 +436,57 @@ public void deleteBy(Object key, Object... keys) {
...
+    }
 
-        @Override
-        public boolean cancel(boolean mayInterruptIfRunning) {
-            boolean result = listenableFuture.cancel(mayInterruptIfRunning);
-            super.cancel(mayInterruptIfRunning);
-            return result;
-        }
+    public ParallelFlux<Row> scanRow(int threads, Select select, Executor executor) {
+        return scan(threads, select()).flatMap(query -> executeFlux(query, executor));
```
Comment (AntonGabov) [inline src/main/java/com/appmetr/cql/CqlDao.java]: Typo? We should use var `select` or call function `select()`?

File: src/main/java/com/appmetr/cql/util/ResultFluxSkeleton.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import com.appmetr.cql.CqlDao;
+import com.datastax.driver.core.PagingIterable;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+
+public class ResultFluxSkeleton<R extends PagingIterable<R, T>, T> {
```
Comment (AntonGabov) [inline src/main/java/com/appmetr/cql/util/ResultFluxSkeleton.java]: Why name contains `Skeleton`? Maybe use better naming

File: src/main/java/com/appmetr/cql/CqlDao.java (hunk 1 above)
Comment (Megaprog) [reply src/main/java/com/appmetr/cql/CqlDao.java]: you are very right

File: src/main/java/com/appmetr/cql/util/ResultFluxSkeleton.java (hunk 2 above)
Comment (Megaprog) [reply src/main/java/com/appmetr/cql/util/ResultFluxSkeleton.java]: Skeleton means каркас. What name will be better?

Comment (AntonGabov) [reply src/main/java/com/appmetr/cql/util/ResultFluxSkeleton.java]: I don't know :) 
As we transfer `result.one` to the `FluxSink` in dedicated thread, so
1) `PagingIterableListener`
2) `PagingIterableEmitter`
3) `PagingIterableProcessor`

Comment (Megaprog) [reply src/main/java/com/appmetr/cql/util/ResultFluxSkeleton.java]: 2.	PagingIterableEmitter sounds good

File: src/main/java/com/appmetr/cql/CqlDao.java [hunk 3]
```diff
@@ -439,45 +435,56 @@ public void deleteBy(Object key, Object... keys) {
...
+            sink.onDispose(fluxSkeleton::dispose);
+        });
     }
 
-    public static <R> CompletableFuture<R> completableFuture(ListenableFuture<R> listenableFuture, Executor executor) {
-        return new CompletableListenable<>(listenableFuture, executor);
+    public Flux<Select.Where> scanQuery(int ranges, Select select) {
+        final long[] tokens = MurMur64Tokens.tokens(ranges);
+        return Flux.range(0, ranges)
+                .map(i -> MurMur64Tokens.whereTokenRange(this, tokens, i, select));
```
Comment (AntonGabov) [inline src/main/java/com/appmetr/cql/CqlDao.java]: Now `scanQuery()` won't be execute in parallel...is it ok behaviour?

Comment (Megaprog) [reply src/main/java/com/appmetr/cql/CqlDao.java]: It is not so expensive to calculate in one thread

**Pull request merged**